From 63bbda58fe6526cbe42d38df7b76ed50594d24a0 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Fri, 23 Jan 2015 17:05:14 +0800
Subject: [PATCH] usb: dwc_otg: handle data toggle error in right way

data toggle error without a ChannelHalt status is not a real data
toggle err, it is a transfer error state so just clear the
interrupt bit rather than resubmit this request

Signed-off-by: lyz <lyz@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_hcd_intr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_hcd_intr.c b/drivers/usb/dwc_otg_310/dwc_otg_hcd_intr.c
index 33018f4f285f..e1dccb1a7a22 100755
--- a/drivers/usb/dwc_otg_310/dwc_otg_hcd_intr.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_hcd_intr.c
@@ -2179,8 +2179,8 @@ int32_t dwc_otg_hcd_handle_hc_n_intr(dwc_otg_hcd_t *dwc_otg_hcd, uint32_t num)
 		    handle_hc_frmovrun_intr(dwc_otg_hcd, hc, hc_regs, qtd);
 	}
 	if (hcint.b.datatglerr) {
-		retval |=
-		    handle_hc_datatglerr_intr(dwc_otg_hcd, hc, hc_regs, qtd);
+		disable_hc_int(hc_regs, datatglerr);
+		clear_hc_int(hc_regs, chhltd);
 	}
 
 	return retval;
-- 
2.35.3

